Tests/function/tests/Remove-VSTeamWiki.Tests.ps1

```diff
@@ -0,0 +1,84 @@
+         $FakeProjectWikiName = 'myWiki2'         
+         
+         # response when using wiki endpoint - deleting code wikis
+         Mock Invoke-RestMethod { Open-SampleFile 'Get-VSTeamWiki.json' -Index 0 } -ParameterFilter { # when wiki is deleted, it returns object as when its created
+            $Method -eq 'DELETE' -and
+            $Uri -like "*wiki/wikis/$($FakeCodeWikiId)*"            
+         }
+         # response when using git endpoint - deleting project wikis
+         Mock Remove-VSTeamGitRepository {return $null} -ParameterFilter {
+            $Id -eq $FakeProjectWikiId
+         } 
+         Mock Get-VSTeamWiki { Open-SampleFile 'Get-VSTeamWiki.json' -ReturnValue } # called with no parameters, return all wikis      
+         Mock Get-VSTeamWiki { Open-SampleFile 'Get-VSTeamWiki.json' -index 0 } -ParameterFilter { # pass project and CODE wiki name returns specific wiki            
+            $Name -eq $FakeCodeWikiName -and
+            $ProjectName -eq $FakeProjectName
+         }
+         Mock Get-VSTeamWiki { Open-SampleFile 'Get-VSTeamWiki.json' -index 1 } -ParameterFilter { # pass project and PROJECT wiki name returns specific wiki            
+            $Name -eq $FakeProjectWikiName -and
+            $ProjectName -eq $FakeProjectName
+         }      
+      }
+      
+      It 'code wiki, by name, should unpublish wiki' {
+         ## Act                  
+         Remove-VSTeamWiki -ProjectName $FakeProjectName -WikiName $FakeCodeWikiName
+         
+         ## Assert
+         Should -Invoke Invoke-RestMethod -Exactly -Times 1 -Scope It -ParameterFilter {
+            $Uri -eq "https://dev.azure.com/$($FakeOrg)/_apis/wiki/wikis/$($FakeCodeWikiId)?api-version=$(_getApiVersion Wiki)"  -and
+            $Method -eq 'DELETE' 
+         }
+      }
+            
+      It 'code wiki, by id, should unpublish wiki' {
+         ## Act
+         Remove-VSTeamWiki -WikiId $FakeCodeWikiId
+         
+         ## Assert
+         Should -Invoke Invoke-RestMethod -Exactly -Times 1 -Scope It -ParameterFilter {
+            $Uri -eq "https://dev.azure.com/$($FakeOrg)/_apis/wiki/wikis/$($FakeCodeWikiId)?api-version=$(_getApiVersion Wiki)"  -and
+            $Method -eq 'DELETE'
+         }
+      }
+
+      It 'project wiki, by name, should unpublish wiki' {
+         ## Act                           
+         Remove-VSTeamWiki -ProjectName $FakeProjectName -WikiName $FakeProjectWikiName
+         
+         ## Assert
+         Should -Invoke Remove-VSTeamGitRepository -Exactly -Times 1 -Scope It -ParameterFilter {
+            $Id -eq $FakeProjectWikiId
+         }
+      }
+            
+      It 'project wiki, by id, should unpublish wiki' {
+         ## Act
+         Remove-VSTeamWiki -WikiId $FakeProjectWikiId
+         
+         ## Assert
+         Should -Invoke Remove-VSTeamGitRepository -Exactly -Times 1 -Scope It -ParameterFilter {
+            $Id -eq $FakeProjectWikiId
+         }
+      }
+   }
```

Comment:
Missing a tests to trigger your exception, when the given wiki is not found. Could you please test the exception as well?

Reply:
added test case for invalid Wiki Id